### DISCUSSION TITLE
Pretty Print storage.general json content

### DISCUSSION THREAD

**Discussion by sseshan7** [zauberzeug/nicegui#3367] [Ideas / Feature Requests]
Hello,

I'd like to add capability to pretty print storage.general json.
Right now, I have to do the following at the very beginning of the entry point to my application.
```py
import orjson
from nicegui.json import orjson_wrapper

orjson_wrapper.ORJSON_OPTS |= orjson.OPT_INDENT_2
```

And if nicegui fallsback to `builtin_wrapper.py`, there's no way to specify an indent, because it is hard-coded to `indent=None`.

**Comment by falkoschindler** [comment]
Hi @sseshan7,

I guess we could introduce an API like this:
```py
from nicegui import ui, app, storage

storage.PersistentDict.default_indent = 4
app.storage.general.indent = 4
app.storage.user.indent = 4
```

What do you think? Would you like to create a pull request?

**Comment by sseshan7** [comment]
That looks good to me! I can create a PR with an implementation and we can continue discussion there.
Would you prefer I close this discussion thread upon creation of the PR?

**Reply by falkoschindler** [reply]
Thanks, @sseshan7! Let's keep this discussion open until the PR is merged. 🙂

**Comment by sseshan7** [comment]
Opened a PR here: https://github.com/zauberzeug/nicegui/pull/3396

Felt a boolean flag was more appropriate, given that `orjson` only supports a 2-space indent, or no indent at all.

**Reply by falkoschindler** [reply]
Oh, I wasn't aware of this limitation. I guess a boolean flag is a reasonable choice.

**Comment by falkoschindler** [comment]
This feature has just been released in version 1.4.30.
